src/cryptonote_core/blockchain.cpp

```diff
@@ -2774,7 +2774,10 @@ bool Blockchain::find_blockchain_supplement(const uint64_t req_start_block, cons
   if(req_start_block > 0)
   {
     // if requested height is higher than our chain, return false -- we can't help
-    if (req_start_block >= m_db->height())
+    uint64_t top_height;
```

Comment:
Nitpick, but you really don't need the temporary `top_height` here or below. Just give `total_height` to `top_block_hash` then `++total_height`.